Flaky e2e test: TestMutationRollingDownscaleCombination

```
=== RUN   TestMutationRollingDownscaleCombination/All_expected_Pods_should_eventually_be_ready#01
Retries (30m0s timeout): ........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
    utils.go:84: 
        	Error Trace:	utils.go:84
        	Error:      	Received unexpected error:
        	            	invalid StatefulSets: expected map[test-combined-upgrade-downscale-grjb-es-data-1:1 test-combined-upgrade-downscale-grjb-es-data-2:1 test-combined-upgrade-downscale-grjb-es-master:1], got map[test-combined-upgrade-downscale-grjb-es-data-1:1 test-combined-upgrade-downscale-grjb-es-data-2:2 test-combined-upgrade-downscale-grjb-es-master:1]
        	Test:       	TestMutationRollingDownscaleCombination/All_expected_Pods_should_eventually_be_ready#01
{"log.level":"error","@timestamp":"2020-09-17T03:50:28.216Z","message":"stopping early","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","error":"test failure","error.stack_trace":"github.com/elastic/cloud-on-k8s/test/e2e/test.StepList.RunSequential\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/step.go:43\ngithub.com/elastic/cloud-on-k8s/test/e2e/test.RunMutations\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/run_mutation.go:38\ngithub.com/elastic/cloud-on-k8s/test/e2e/test.RunMutation\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/run_mutation.go:80\ngithub.com/elastic/cloud-on-k8s/test/e2e/es.RunESMutation\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/es/mutation_test.go:299\ngithub.com/elastic/cloud-on-k8s/test/e2e/es.TestMutationRollingDownscaleCombination\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/es/mutation_test.go:187\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1127"}
    --- FAIL: TestMutationRollingDownscaleCombination/All_expected_Pods_should_eventually_be_ready#01 (1800.00s)
```

https://devops-ci.elastic.co/job/cloud-on-k8s-e2e-tests-kind-k8s-versions/213/testReport/

--- a/test/e2e/es/mutation_test.go
+++ b/test/e2e/es/mutation_test.go
@@ -174,12 +174,12 @@ func TestMutationSecondMasterSetDown(t *testing.T) {
 func TestMutationRollingDownscaleCombination(t *testing.T) {
 	b := elasticsearch.NewBuilder("test-combined-upgrade-downscale").
 		WithESMasterNodes(1, elasticsearch.DefaultResources).
-		WithNamedESDataNodes(1, "data-1", elasticsearch.DefaultResources).
+		WithNamedESDataNodes(2, "data-1", elasticsearch.DefaultResources).
 		WithNamedESDataNodes(2, "data-2", elasticsearch.DefaultResources)
 
 	mutated := b.WithNoESTopology().
 		WithESMasterNodes(1, elasticsearch.DefaultResources).
-		WithNamedESDataNodes(1, "data-1", elasticsearch.DefaultResources).
+		WithNamedESDataNodes(2, "data-1", elasticsearch.DefaultResources).
 		WithNamedESDataNodes(1, "data-2", elasticsearch.DefaultResources). // scaling down data-2
 		WithAdditionalConfig(map[string]map[string]interface{}{
 			"data-1": {

